util/log/file.go

```diff
@@ -85,36 +97,92 @@ func shortHostname(hostname string) string {
 	return hostname
 }
 
-// logName returns a new log file name containing tag, with start time t, and
-// the name for the symlink for tag.
-func logName(tag string, t time.Time) (name, link string) {
-	name = fmt.Sprintf("%s.%s.%s.log.%s.%04d%02d%02d-%02d%02d%02d.%d",
-		program,
-		host,
-		userName,
-		tag,
-		t.Year(),
-		t.Month(),
-		t.Day(),
-		t.Hour(),
-		t.Minute(),
-		t.Second(),
+// escapeString replaces all the periods in the string with an underscore, and
+// every underscore with a double underscore.
+func escapeStringForFilename(s string) string {
+	sEscapedPartial := strings.Replace(s, "_", "__", -1)
+	return strings.Replace(sEscapedPartial, ".", "_", -1)
+}
+
+// unescapeStringForFilename reverts the escaping from escapeStringForFilename.
+func unescapeStringForFilename(s string) string {
+	sUnescapedPartial := strings.Replace(s, "_", ".", -1)
+	return strings.Replace(sUnescapedPartial, "__", "_", -1)
+}
+
+// logName returns a new log file name containing the severity, with start time
+// t, and the name for the symlink for the severity.
+func logName(severity Severity, t time.Time) (name, link string) {
+	// Replace the ':'s in the time format with '_'s to allow for log files in
+	// Windows.
+	tFormatted := strings.Replace(t.Format(time.RFC3339), ":", "_", -1)
+
+	name = fmt.Sprintf("%s.%s.%s.log.%s.%s.%d",
+		escapeStringForFilename(program),
+		escapeStringForFilename(host),
+		escapeStringForFilename(userName),
+		severity.Name(),
+		tFormatted,
 		pid)
-	return name, program + "." + tag
+	return name, program + "." + severity.Name()
+}
+
+// A FileDetails holds all of the particulars that can be parsed by the name of
+// a log file.
+type FileDetails struct {
+	Program  string
+	Host     string
+	UserName string
+	Severity Severity
+	Time     time.Time
+	PID      int
+}
+
+func parseLogFilename(filename string) (FileDetails, error) {
+	matches := logFileRE.FindStringSubmatch(filename)
+	if matches == nil || len(matches) != 7 {
+		return FileDetails{}, util.Errorf("not a log file")
+	}
+
+	sev, sevFound := SeverityByName(matches[4])
+	if !sevFound {
```

Comment:
Sorry, I don't understand. Are you saying the error string is inspected programmatically somewhere?